src/Servers/IIS/IntegrationTesting.IIS/src/IISDeployer.cs

```diff
@@ -473,6 +490,18 @@ private static void RetryServerManagerAction(Action<ServerManager> action)
             delay *= 1.5;
         }
 
+        // Try to upload the applicationHost config on helix to help debug
+        UploadFileOnHelix(_applicationHostConfig, "serverManagerRetryFailed.applicationHost.config");
+
         throw new AggregateException($"Operation did not succeed after {retryCount} retries", exceptions.ToArray());
     }
+
+    private static void UploadFileOnHelix(string filePath, string uploadFileName)
```

Comment:
Might be useful to have this exposed generally somewhere since this is pretty handy for debugging helix failures

Reply:
Yes to exposing the method or something like `static UploadConfigFiles(string applicationRootPath, string uploadFilePrefix)`. Suggest sticking it in Microsoft.AspNetCore.Testing, Microsoft.AspNetCore.Server.IntegrationTesting, or Microsoft.AspNetCore.Server.IntegrationTesting.IIS, depending mostly on just how general the method feels to you.

Separately, this method doesn't upload the files just on Helix. Suggest
```suggestion
    public static void UploadOrSafeToTempFile(string filePath, string uploadFileName) { ... }
```

Reply:
Added a new HelixHelper for this and some of the other utility code that was living in the SkipOnHelix attribute, went with HelixHelper.PreserveFile which will either upload/copy to temp